crates/typst-pdf/src/page.rs

```diff
@@ -141,6 +142,18 @@ pub(crate) fn write_page_tree(ctx: &mut PdfContext) {
     ctx.colors.write_functions(&mut ctx.pdf);
 }
 
+fn name_from_loc<'a>(doc: &Document, loc: &Location) -> Option<Name<'a>> {
+    let mut query_res = doc.introspector.query(&Selector::Location(*loc));
```

Comment:
This could probably use query_first and then we can skip the assertions below. However, does this really account for the situation where this isn't the first heading with the label?

Reply:
The function assumes that the heading is referenceable, which requires the label to be unique, so the case you are describing can't occur.
The second assertion ensures that our assumption holds.


I wasn't sure that a Location identifies exactly one element, so I put in my first assertion. But this is the case (right?) and can be removed.
I think, the only reason I didn't use `query_first` was to be able to write down this assertion.

Reply:
Locations are indeed guaranteed to be unique